### ISSUE TITLE
The AUR should not be a supported platform

### ISSUE THREAD

**Issue by dvzrv** [libguestfs/supermin#20]
Hi! :wave: 

I package libguestfs for Arch Linux. When investigationg https://github.com/libguestfs/libguestfs/issues/139 I noticed the following:

```
--2024-04-15 14:38:02--  https://aur.archlinux.org/packages/cd/cdrtools/cdrtools.tar.gz
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving aur.archlinux.org (aur.archlinux.org)... 95.216.144.15, 2a01:4f9:c010:50::1
Connecting to aur.archlinux.org (aur.archlinux.org)|95.216.144.15|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-04-15 14:38:02 ERROR 404: Not Found.

supermin: set -e
          umask 0000
          cd '/var/tmp/supermind89798.tmpdir/ytx0efty'
          wget 'https://aur.archlinux.org/packages/cd/cdrtools/cdrtools.tar.gz'
          tar xf 'cdrtools.tar.gz'
          cd 'cdrtools'
          /usr/bin/makepkg
          mv 'cdrtools'-*.pkg.tar.xz '/var/tmp/supermind89798.tmpdir/ytx0efty'
       : command failed, see earlier errors
make[2]: *** [Makefile:1080: stamp-supermin] Error 1
make[2]: Leaving directory '/build/libguestfs/src/libguestfs-1.52.0/appliance'
make[1]: *** [Makefile:1089: all-recursive] Error 1
make[1]: Leaving directory '/build/libguestfs/src/libguestfs-1.52.0'
make: *** [Makefile:995: all] Error 2
```

This appears to be implemented in https://github.com/libguestfs/supermin/blob/5a44ffc68f6eb2906e761dad63dcd207b5ca607a/src/ph_pacman.ml#L196-L234

Frankly, this is quite the horrific approach. The AUR is unsupported for a reason and should *never* be used in the context of distribution packages.
In this particular case supermin attempts to build a package that is available in the repositories ([cdrtools](https://archlinux.org/packages/extra/x86_64/cdrtools/)), but not in the AUR.

If the packages can not be retrieved, the build process should *not* fall back to building from unverified and untrusted sources but instead just fail!
To be more specific: Please do not automatically build from the AUR at all!

**Comment by rwmjones** [comment]
Sure, we can drop AUR if you like.  Are there other packages which are no in the base Arch distribution that we need?

**Comment by dvzrv** [comment]
That's a good question. Where do I need to look to find out?

Are these the relevant blocks?
https://github.com/libguestfs/libguestfs/blob/6ba64125d960c820b75727e3856b0df0ed6b264c/appliance/packagelist.in#L99-L123
https://github.com/libguestfs/libguestfs/blob/6ba64125d960c820b75727e3856b0df0ed6b264c/appliance/packagelist.in#L243-L298

I know that e.g. zerofree is not packaged for the official repos, but it doesn't appear to be built from the AUR (see https://gitlab.archlinux.org/archlinux/packaging/packages/libguestfs/-/issues/1).

Could you explain which of the packages are hard dependencies?

**Comment by rwmjones** [comment]
Yes it's those.  Supermin actually ignores packages that it doesn't know about (by design), so it's rather hard to know which ones are really required and which aren't.  A number of them will just enable optional features in libguestfs but it'll work without them.

zerofree is definitely not important.

I was just wonder if there are major packages in AUR which are not in base Arch that we should know about, however if not I guess we can just try it.

**Comment by dvzrv** [comment]
> Yes it's those.

Just to make sure: Are all those pkgs only build time dependencies?
With libguestfs I always had a hard time figuring out what is build time and what is runtime dependency :S

**Comment by dvzrv** [comment]
The following dependencies do not exist (neither in the official repositories nor in the AUR):
* ntfs-3g-system-compression
* gostsum
* module-init-tools
* procps (we have procps-ng though)
* util-linux-ng (we have util-linux though and I have never heard of util-linux-ng)
* exfat-fuse
* fuse-exfat

The following only exist in the AUR:
* scrub
* zerofree

**Comment by rwmjones** [comment]
> > Yes it's those.
> 
> Just to make sure: Are all those pkgs only build time dependencies? With libguestfs I always had a hard time figuring out what is build time and what is runtime dependency :S

They're both.  Libguestfs builds a skeleton appliance and fills it in with files from the host filesystem at runtime (https://rwmj.wordpress.com/2014/03/08/supermin-version-5/).

Looking at the list above, I would say the only one which is important is util-linux{-ng}.  I'm sure you must be packaging util-linux-ng, but probably it's been renamed as util-linux.  Old util-linux was deprecated years ago. (https://en.wikipedia.org/wiki/Util-linux)
